src/internal/connector/graph/service.go

```diff
@@ -253,6 +253,14 @@ func GetMiddlewares(maxRetry int, delay time.Duration) []khttp.Middleware {
 		&ThrottleControlMiddleware{},
 		&MetricsMiddleware{},
 	}
+
+	// Optionally add concurrency limiter middleware if it has been initialized
+	cl := GetConcurrencyLimiter()
```

Comment:
nit: this should be moved to the top.  It feels a bit out of order to save a prepend for the last step.  Something like:
```go
func GetMiddlewares(...) {
  mw := []khttp.Middleware{}

  if cl != nil {
    mw = append(mw, cl)
  }

  mw = append(mw, theRest...)
  return mw
}
```

Reply:
Good catch!